// Copyright (c) 2024 Advanced Micro Devices, Inc. All Rights Reserved.
syntax = "proto3";

package Schola;

message BoxPoint {
    repeated float values = 1;
}

message DiscretePoint {
    repeated int32 values = 1;
}

message BinaryPoint {
    repeated bool values = 1;
}

message FundamentalPoint {
    oneof point {
        BoxPoint box_point = 1;
        DiscretePoint discrete_point = 2;
        BinaryPoint binary_point = 3;
    }
}

message DictPoint {
   //This will be converted to a Dict on the python side
   repeated FundamentalPoint values = 1;
}